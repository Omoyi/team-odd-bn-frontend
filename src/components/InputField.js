import React from 'react';
import dotenv from 'dotenv';
import '../assets/css/signup.scss';
import '../assets/css/signin.scss';
import propTypes from 'prop-types';

dotenv.config();
const singleInput = {};
const InputField = ({ handleChange, inputList }) => inputList.map((i) => (
  <div className={i.className} key={i.input}>
    <div>{i.input}</div>
    <input
      required
      pattern={i.pattern}
      title={i.title}
      type={i.type}
      id={i.className}
      onChange={handleChange ? ((e) => {
        singleInput[i.className] = e.target.value;
        return handleChange(singleInput);
      }) : () => 0}
    />
  </div>
));
InputField.propTypes = {
  handleChange: propTypes.func,
  className: propTypes.array,
};
InputField.defaultProps = {
  handleChange: null,
  inputList: null,
};
export default InputField;
