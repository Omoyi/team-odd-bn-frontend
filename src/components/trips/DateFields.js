import React from 'react';
import '../../assets/css/base.scss';
import '../../assets/css/roundtrip.scss';

const inputValues = {};
const checkCurrentDate = () => {
  const currentDate = new Date();
  return JSON.stringify(currentDate).slice(1, 11);
};
const DateFields = ({ dateFieldName, handleChange }) => dateFieldName.map((x) => (

  <div className={x.className} key={x.className}>
    <label htmlFor={x.htmlForName}>{ x.FieldName }</label>
    <input
      data_test={x.data_test}
      type={x.type}
      name={x.name}
      required
      min={checkCurrentDate()}
      id={x.id}
      value={x.value}
      onChange={(e) => {
        inputValues[x.name] = e.target.value;
        return handleChange(inputValues);
      }}
    />
  </div>

));

export default DateFields;
