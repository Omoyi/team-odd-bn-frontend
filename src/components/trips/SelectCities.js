import React from 'react';
import '../../assets/css/base.scss';

const inputsValues = {};
const SelectCities = ({ myData, handleChange, inputName }) => inputName.map((a) => (
  <div className={a.className} key={a.className}>
    <select
      onChange={(e) => {
        inputsValues[a.name] = e.target.value;
        return handleChange(inputsValues);
      }}
      defaultValue=""
      required
    >
      <option value="" disabled>
        { a.PlaceHolderName}
      </option>

      {myData
      && myData.map((cities) => (
        <option name={a.name} value={cities.id} key={cities.id}>
          {cities.city}
        </option>
      ))}
    </select>
  </div>
));

export default SelectCities;
