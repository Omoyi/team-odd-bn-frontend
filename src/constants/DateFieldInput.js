export const onewayDateInputs = [
  {
    type: 'date', name: 'startDate', data_test: 'startDate', FieldName: 'Start date',
  },
];

const dateInputs = [
  {
    className: 'startingDate', type: 'date', name: 'startDate', data_test: 'startDate', FieldName: 'Start date',
  },
  {
    className: 'returningDate', type: 'date', name: 'returnDate', data_test: 'returnDate', FieldName: 'Return date',
  },
];

export default dateInputs;
