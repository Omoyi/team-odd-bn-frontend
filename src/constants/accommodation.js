
export const createAccommodationList = [
  {
    input: 'Name',
    className: 'name',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Google coordinates',
    className: 'googleCoordinates',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Address',
    className: 'address',
    type: 'text',
    title: 'address should be valid',
  },
];
export const modalInput = [
  {
    input: 'Room name',
    className: 'room-name',
    type: 'text',
    title: 'Room name should be valid',
  },
];
