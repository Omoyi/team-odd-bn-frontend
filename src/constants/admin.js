import React from 'react';
import { Link, BrowserRouter as Router } from 'react-router-dom';
import { connect } from 'react-redux';
import { handleFilter } from '../helpers/filter';
import { checkPrevilege } from '../helpers/checkPrevilege';
import usersIcon from '../assets/images/users.png';

export const tableHeads = [
  { className: 'fNameNav', attribute: 'First name', key: 'firstName' },
  { className: 'lNameNav', attribute: 'Last name', key: 'lastName' },
  { className: 'emailNav', attribute: 'Email', key: 'email' },
  { className: 'role', attribute: 'Role', key: 'role' },
  { className: 'EditRoleNav', attribute: 'Edit role', key: 'editRole' },
];
export const userRoleNav = (data) => {
  let newData;
  const roleNavs = [
    {
      className: 'tableHeader',
      attribute: <h2>Assign Roles</h2>,
      key: 'tableHeader',
    },
  ];
  return { roleNavs, newData };
};

export const usersNav = () => checkPrevilege(1) && (
<div>
  <img src={usersIcon} alt="Users icon" />
  <span>Users</span>
</div>
);
export const mapStateToProps = (state) => ({
  admin: state.allUsers.users,
  pageNo: state.pagination.pageNo,
  itemsPerPage: state.pagination.itemsPerPage,
});

export default connect(mapStateToProps, {})(userRoleNav);
