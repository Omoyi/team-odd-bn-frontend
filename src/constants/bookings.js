import React from 'react';

export default (
  availlableRoomTypes,
  accommodationId,
  availlableRooms,
  roomId,
  updateBookAccommodationInput,
  roomType,
) => [
  {
    label: 'Room type',
    inputField:
  <select
    key="selectRoomType"
    id="selectRoomType"
    onChange={(e) => updateBookAccommodationInput({ roomType: e.target.value })}
    disabled={!accommodationId}
  >
    <option>Select room type</option>
    {availlableRoomTypes
       && availlableRoomTypes.map((room) => (
         <option value={room} key={room}>
           {room}
         </option>
       ))}
  </select>,
  },
  {
    label: 'Room name',
    inputField:
  <select
    key="selectRoomName"
    id="selectRoomName"
    onChange={(e) => updateBookAccommodationInput({ roomId: parseInt(e.target.value, 10) })}
    disabled={!roomType}
  >
    <option>Select room name</option>
    {availlableRooms
       && availlableRooms.map((room) => (
         <option value={room.id} key={room.name}>
           {room.name}
         </option>
       ))}
  </select>,
  },
  {
    label: 'Check-in date',
    inputField: <input
      key="checkInDate"
      id="checkInDate"
      onChange={(e) => updateBookAccommodationInput({ checkInDate: e.target.value })}
      disabled={!roomId}
      type="date"
    />,
  },
  {
    label: 'Check-out date',
    inputField: <input
      key="checkOutDate"
      id="checkOutDate"
      onChange={(e) => updateBookAccommodationInput({ checkOutDate: e.target.value })}
      disabled={!roomId}
      type="date"
    />,
  },
];
