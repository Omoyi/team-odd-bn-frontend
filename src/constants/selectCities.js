const selectInputs = [
  {
    className: 'originCity', name: 'originId', PlaceHolderName: 'Your Origin', value: 'originId',
  },
  {
    className: 'destinationCity', name: 'destinationId', PlaceHolderName: 'Your Destination', value: 'destinationId',
  },
];

export default selectInputs;
