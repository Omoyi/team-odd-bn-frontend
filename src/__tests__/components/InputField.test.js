import React from 'react';
import { shallow } from 'enzyme';
import InputField from '../../components/InputField';
import { signupInputs } from '../../constants/signUp';
import loginInputs from '../../constants/login';
import SelectCities from '../../components/trips/SelectCities';
import selectInputs from '../../constants/selectCities';
import { onewayDateInputs } from '../../constants/DateFieldInput';
import DateFields from '../../components/trips/DateFields';

describe('First React component test with Enzyme', () => {
  it('should renders Date Input component', () => {
    expect(shallow(<DateFields dateFieldName={onewayDateInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <DateFields
        dateFieldName={[{
          type: 'date', name: 'startDate', data_test: 'startDate', FieldName: 'Start date',
        }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('input');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ startDate: 'matched' });
  });
  it('should renders Select Input component', () => {
    expect(shallow(<SelectCities inputName={selectInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <SelectCities
        inputName={[{
          className: 'originCity', name: 'originId', PlaceHolderName: 'Your Origin', value: 'originId',
        }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('select');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ originId: 'matched' });
  });
  it('should call onChange prop for Login page', () => {
    expect(shallow(<InputField inputList={loginInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <InputField
        value=""
        placeholder=""
        inputList={[{ input: 'matched', className: 'matched' }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('input');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ matched: 'matched' });
  });

  it('should call onChange prop for signup', () => {
    expect(shallow(<InputField inputList={signupInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <InputField
        value=""
        placeholder=""
        inputList={[{ input: 'matched', className: 'matched' }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('input');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ matched: 'matched' });
  });
});
