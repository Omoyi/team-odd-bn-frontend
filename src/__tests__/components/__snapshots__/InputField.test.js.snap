// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme should call onChange prop for Login page 1`] = `ShallowWrapper {}`;

exports[`First React component test with Enzyme should call onChange prop for signup 1`] = `ShallowWrapper {}`;

exports[`First React component test with Enzyme should renders Date Input component 1`] = `ShallowWrapper {}`;

exports[`First React component test with Enzyme should renders Select Input component 1`] = `ShallowWrapper {}`;
